uuid.lua

```diff
@@ -0,0 +1,31 @@
+--[[
+    uuid generation
+]]
+
+local path = (...):gsub("uuid", "")
+
+local uuid = {}
+
+--helper for optionally passed random; defaults to love.math.random if present, otherwise math.random
+local _global_random = math.random
+if love and love.math and love.math.random then
+	_global_random = love.math.random
+end
+
+local function _random(min, max, r)
```

Comment:
so the idea when using this is to allow the user to pass in their own rng object; there doesn't seem to be a provided way to do that here?